vllm/distributed/utils.py

```diff
@@ -46,3 +55,60 @@ def split_tensor_along_last_dim(
         return tuple(chunk.contiguous() for chunk in tensor_list)
 
     return tensor_list
+
+
+# code partly borrowed from
+# https://github.com/turboderp/exllamav2/blob/1c67f97f3d2a968605a9c31ab791a05c85bb7879/exllamav2/compat.py#L10
+# License: MIT
+def _can_actually_p2p(idx_a, idx_b):
+    dev_i = f"cuda:{idx_a}"
+    dev_j = f"cuda:{idx_b}"
+    a = torch.randn(5, device=dev_i) + 123.0
+    b = a.to(dev_j)
+    c = b.to(dev_i)
+    return torch.all(a == c).cpu().item()
+
+
+_gpu_p2p_access_cache = None
```

Comment:
add type?

Reply:
fixed in [72e1128](https://github.com/vllm-project/vllm/pull/4021/commits/72e1128965ea30f7d48256ab8ccb3b4547f60de8).